### ISSUE TITLE
Overflow on absolute comparison of signed integers

### ISSUE THREAD

**Issue by pcpthm** [brendanzab/approx#52]
```rust
fn main() {
    approx::abs_diff_ne!(std::i32::MAX, std::i32::MIN);
}
```
Expected behavior: Runs fine.
Actual behavior: `thread 'main' panicked at 'attempt to subtract with overflow'`.

**Comment by pcpthm** [comment]
More corner case:
```rust
approx::abs_diff_ne!(-1, std::i32::MAX)
```
won't overflow on subtraction but will be overflow on `abs`

**Comment by brendanzab** [comment]
Oh huh, that's interesting. We actually implement these comparisons for integers?

**Comment by brendanzab** [comment]
Right, seems like it's here: https://github.com/brendanzab/approx/blob/0527afee5faf712e877634ccf4b9790cac5a1d63/src/abs_diff_eq.rs#L35-L87
